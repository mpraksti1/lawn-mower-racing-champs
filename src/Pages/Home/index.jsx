
import React from 'react';
// import SpeedIcon from 'react-icons/lib/md/update';
// import CommunityIcon from 'react-icons/lib/md/people';
import DangerIcon from 'react-icons/lib/md/error';
import Card from '../../Components/Card';
import Text from '../../Elements/Text';
import Hero from '../../Components/Hero';
import StyledDiv from './styles';

const homeItems = [{
  title: 'Speed!',
  copy: 'The fastest lawn mowers this side of the river, guarunteed.',
  url: '/assets/images/home-item-1.jpg',
}, {
  title: 'Community!',
  copy: 'The largest group of LMRC enthusiests online.',
  url: '/assets/images/home-item-2.jpg',
}, {
  title: 'Danger!',
  copy: 'This is full contact Lawn mower racing at it\'s best.',
  url: '/assets/images/home-item-3.jpg',
}];

const Home = () => (
  <div>
    <Hero>
      <Text sans xxlg spaceBelow block>Welcome to the Lawn Mower Racing Championships!</Text>
      <Text sans thin lg block>
        Add and remove racers from the roster at will, their fate is in your hands.
      </Text>
    </Hero>
    <StyledDiv>
      {homeItems.map(homeItem => (
        <Card key={homeItem.title} title={homeItem.title} copy={homeItem.copy} bg={homeItem.url}>
          <DangerIcon />
        </Card>
      ))}
    </StyledDiv>
  </div>
);

export default Home;
