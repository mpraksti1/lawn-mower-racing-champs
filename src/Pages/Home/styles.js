
import styled from 'styled-components';

const StyledDiv = styled.div`
  display: grid;
  grid-column-gap: 1px;
  grid-row-gap: 1px;
  margin: ${({ theme }) => theme.baseSizeUnit * 2}px 0 50px;
  background: linear-gradient(
    to right,
    ${({ theme }) => theme.colors.green},
    ${({ theme }) => theme.colors.yellow}
  );
  padding: 2px;

  @media screen and (min-width: 768px) {
    grid-template-columns: 1fr 1fr 1fr;
  }

  div {
    padding: ${({ theme }) => theme.baseSizeUnit * 2}px;
  }
`;

export default StyledDiv;
