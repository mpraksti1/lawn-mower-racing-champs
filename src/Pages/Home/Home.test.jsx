import Home from '.';

describe('Home page component', () => {
  it('renders without crashing', () => {
    shallow(<Home />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });
});
