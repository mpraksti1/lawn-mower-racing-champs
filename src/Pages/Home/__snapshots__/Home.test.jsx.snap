// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component matches snapshot 1`] = `
<div>
  <Hero
    imgUrl=""
  >
    <Text
      block={true}
      sans={true}
      spaceBelow={true}
      xxlg={true}
    >
      Welcome to the Lawn Mower Racing Championships!
    </Text>
    <Text
      block={true}
      lg={true}
      sans={true}
      thin={true}
    >
      Add and remove racers from the roster at will, their fate is in your hands.
    </Text>
  </Hero>
  <styled.div>
    <Card
      bg="/assets/images/home-item-1.jpg"
      copy="The fastest lawn mowers this side of the river, guarunteed."
      key="Speed!"
      title="Speed!"
    >
      <MdError />
    </Card>
    <Card
      bg="/assets/images/home-item-2.jpg"
      copy="The largest group of LMRC enthusiests online."
      key="Community!"
      title="Community!"
    >
      <MdError />
    </Card>
    <Card
      bg="/assets/images/home-item-3.jpg"
      copy="This is full contact Lawn mower racing at it's best."
      key="Danger!"
      title="Danger!"
    >
      <MdError />
    </Card>
  </styled.div>
</div>
`;
